src/main/java/uk/gov/justice/digital/delius/service/ConvictionService.java

```diff
@@ -132,6 +146,28 @@ public Conviction addCourtCaseFor(Long offenderId, CourtCase courtCase) {
                 throw new DuplicateConvictionsForBookingNumberException(events.size());
         }
     }
+    public Optional<Event> getSingleActiveConvictionIdByOffenderIdAndCloseToSentenceDate(Long offenderId, LocalDate sentenceStartDate) throws DuplicateConvictionsForSentenceDateException {
+        val events = eventRepository.findByOffenderIdWithCustody(offenderId)
+                .stream()
+                .filter(event -> event.getActiveFlag() == 1L)
+                .filter(event -> didSentenceStartAroundDate(event, sentenceStartDate))
+                .collect(toList());
+
+        switch (events.size()) {
+            case 0:
+                return Optional.empty();
+            case 1:
+                return events.stream().findFirst();
+            default:
+                throw new DuplicateConvictionsForSentenceDateException(events.size());
+        }
+    }
+
+    private boolean didSentenceStartAroundDate(Event event, LocalDate sentenceStartDate) {
+        // typically used to match start dates in NOMIS and Delius which may be out by a few days
+        return DAYS.between(event.getDisposal().getStartDate(), sentenceStartDate) <= 7;
```

Comment:
Further analysis required to see if 7 days is the correct leniency between NOMIS and Delius data

Reply:
Should 7 be a constant?

Reply:
Good point, I'll do that next PR